mysql/gcpmysql/gcpmysql.go

```diff
@@ -113,18 +113,35 @@ func (uo *URLOpener) OpenMySQLURL(ctx context.Context, u *url.URL) (*sql.DB, err
 	mysql.RegisterDial(dialerName, client.Dial)
 
 	password, _ := u.User.Password()
-	cfg := &mysql.Config{
-		AllowNativePasswords: true,
-		Net:                  dialerName,
-		Addr:                 instance,
-		User:                 u.User.Username(),
-		Passwd:               password,
-		DBName:               dbName,
+
+	cfg, err := configWithOptions(u)
+	if err != nil {
+		return nil, fmt.Errorf("gcpmysql: open config %v", err)
 	}
+
+	cfg.AllowNativePasswords = true
```

Comment:
I think the tests for this would be more clear and more end-to-end if this code moved into `configWithOptions` (which maybe then should be renamed `configFromURL` or something).